Vic2ToHoI4/Source/HOI4World/HoI4Country.cpp

```diff
@@ -276,6 +277,182 @@ void HoI4::Country::convertLeaders(const graphicsMapper& theGraphics, const Vic2
+					  " Dominions und ist vereint und stolz auf sein kaiserliches Erbe.",
+				 "german");
+			hoi4Localisations.addIdeaLocalisation(tag + "_monarch_desc",
+				 "Gromadząc się wokół Króla " + *name + " i " + *adjective + " Dominium, " + *adjective +
+					  " są zjednoczeni i dumni ze swojego imperialnego dziedzictwa.",
+				 "polish");
+			hoi4Localisations.addIdeaLocalisation(tag + "_monarch_desc",
+				 "Сплотившись вокруг Короля " + *name + " и " + *adjective + " Доминионов, " + *adjective +
+					  " народ объединяется и гордится своим имперским наследием.",
+				 "russian");
+			hoi4Localisations.addIdeaLocalisation(tag + "_monarch_desc",
+				 "Al reunirse en torno al Rey de" + *name + " y los Dominios " + *adjective + ", el pueblo " + *adjective +
+					  " está unido y orgulloso de su legado imperial.",
+				 "spanish");
+		}
+		else
+		{
+			hoi4Localisations.addIdeaLocalisation(tag + "_monarch_desc",
+				 "Rallying around the Queen of " + *name + " and the " + *adjective + " Dominions, the " + *adjective +
+					  " people stand united and proud of their imperial legacy.",
+				 "english");
+			hoi4Localisations.addIdeaLocalisation(tag + "_monarch_desc",
+				 "Reunindo-se em torno da Rainha da " + *name + " e dos Domínios " + *adjective + ", o povo " + *adjective +
+					  " se mantém unido e orgulhoso de seu legado imperial.",
+				 "braz_por");
+			hoi4Localisations.addIdeaLocalisation(tag + "_monarch_desc",
+				 "Rassemblant autour de la reine de " + *name + " et des dominions " + *adjective + " le peuple " +
+					  *adjective + " est uni et fier de son héritage impérial.",
+				 "french");
+			hoi4Localisations.addIdeaLocalisation(tag + "_monarch_desc",
+				 "Das " + *adjective + " Volk versammelt sich um die Königin von " + *name + " und die " + *adjective +
+					  " Dominions und ist vereint und stolz auf sein kaiserliches Erbe.",
+				 "german");
+			hoi4Localisations.addIdeaLocalisation(tag + "_monarch_desc",
+				 "Zbierając się wokół Królowej" + *name + " i " + *adjective + " Dominium, " + *adjective +
+					  " są zjednoczeni i dumni ze swojego imperialnego dziedzictwa.",
+				 "polish");
+			hoi4Localisations.addIdeaLocalisation(tag + "_monarch_desc",
+				 "Сплотившись вокруг Королевы " + *name + " и " + *adjective + " Доминионов, " + *adjective +
+					  " народ объединяется и гордится своим имперским наследием.",
+				 "russian");
+			hoi4Localisations.addIdeaLocalisation(tag + "_monarch_desc",
+				 "Reunidos en torno a la Reina de" + *name + " y los Dominios " + *adjective + ", el pueblo " + *adjective +
+					  " está unido y orgulloso de su legado imperial.",
+				 "spanish");
+		}
+	}
+
+	if (!female)
+	{
+		const auto monarchPortraits = theGraphicsmapper.getMaleMonarchPortraits(primaryCultureGroup);
+		if (!monarchPortraits.empty())
+		{
+			monarchIdeaTexture = monarchPortraits.at(
+				 std::uniform_int_distribution<int>{0, static_cast<int>(monarchPortraits.size() - 1)}(generator));
```

Comment:
Could you not use random for this? Rather some cycling method that would run through all available portraits instead of assigning same one to 3 countries if there are indeed 2+ portraits available. Just a QoL thing.

Reply:
I'll look into that kind of improvement as I start cleaning up HoI4 world. For now, its the same as other portraits and some names.